### PR TITLE
Issue #22

### PR DESCRIPTION
**Pull request by william-wc**

### CODE REVIEW

File: core/src/servico/NotificacaoService.java [hunk 1]
```diff
@@ -45,38 +45,25 @@ public String notificarAusencia(Long idProfessor, String dataInicio, String data
...
-            inícioReal = new DateTime(inicio.getYear(),inicio.getMonthOfYear(),inicio.getDayOfMonth(),inicio.getHourOfDay(),inicio.getMinuteOfHour());
-
             fim = new DateTime(sdf.parse(dataFim));
-            fimReal = new DateTime(fim.getYear(),fim.getMonthOfYear(),fim.getDayOfMonth(),fim.getHourOfDay(),fim.getMinuteOfHour());
+        } catch (ParseException pe) {
+            try {
+                sdf = new SimpleDateFormat("dd/MM/yyyy");
+                inicio = new DateTime(sdf.parse(dataInicio)).withHourOfDay(0).withMinuteOfHour(0);
+                fim = new DateTime(sdf.parse(dataFim)).withHourOfDay(23).withMinuteOfHour(59);
+            } catch (ParseException pe2) {
```
Comment (calebepb) [inline core/src/servico/NotificacaoService.java]: Codacy Issue found: [A catch statement that catches an exception only to rethrow it should be avoided.](https://www.codacy.com/app/calebepb/substituicao/file/2734585602/issues/source?bid=3308597&fileBranchId=3351375#l59)

File: core/src/servico/NotificacaoService.java [hunk 2]
```diff
@@ -45,38 +45,25 @@ public String notificarAusencia(Long idProfessor, String dataInicio, String data
...
-                        
             inicio = new DateTime(sdf.parse(dataInicio));
-            inícioReal = new DateTime(inicio.getYear(),inicio.getMonthOfYear(),inicio.getDayOfMonth(),inicio.getHourOfDay(),inicio.getMinuteOfHour());
-
             fim = new DateTime(sdf.parse(dataFim));
-            fimReal = new DateTime(fim.getYear(),fim.getMonthOfYear(),fim.getDayOfMonth(),fim.getHourOfDay(),fim.getMinuteOfHour());
+        } catch (ParseException pe) {
+            try {
+                sdf = new SimpleDateFormat("dd/MM/yyyy");
+                inicio = new DateTime(sdf.parse(dataInicio)).withHourOfDay(0).withMinuteOfHour(0);
```
Comment (calebepb) [inline core/src/servico/NotificacaoService.java]: Codacy Issue found: [Line is longer than 80 characters (found 98).](https://www.codacy.com/app/calebepb/substituicao/file/2734585602/issues/source?bid=3308597&fileBranchId=3351375#l57)

File: core/src/servico/NotificacaoService.java [hunk 3]
```diff
@@ -45,38 +45,25 @@ public String notificarAusencia(Long idProfessor, String dataInicio, String data
...
+                fim = new DateTime(sdf.parse(dataFim)).withHourOfDay(23).withMinuteOfHour(59);
```
Comment (calebepb) [inline core/src/servico/NotificacaoService.java]: Codacy Issue found: ['23' is a magic number.](https://www.codacy.com/app/calebepb/substituicao/file/2734585602/issues/source?bid=3308597&fileBranchId=3351375#l58)

**Comment by calebepb** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 6
- Added 3
           
```


See the complete overview on [Codacy](https://www.codacy.com/app/calebepb/substituicao/pullRequest?prid=254847&bid=3351375)

File: core/src/servico/NotificacaoService.java (hunk 1 above)
Comment (calebepb) [inline core/src/servico/NotificacaoService.java]: Codacy Issue found: [A catch statement that catches an exception only to rethrow it should be avoided.](https://www.codacy.com/app/calebepb/substituicao/file/2734585602/issues/source?bid=3308597&fileBranchId=3351375#l59)

File: core/src/servico/NotificacaoService.java (hunk 2 above)
Comment (calebepb) [inline core/src/servico/NotificacaoService.java]: Codacy Issue found: [Line is longer than 80 characters (found 98).](https://www.codacy.com/app/calebepb/substituicao/file/2734585602/issues/source?bid=3308597&fileBranchId=3351375#l57)

File: core/src/servico/NotificacaoService.java (hunk 3 above)
Comment (calebepb) [inline core/src/servico/NotificacaoService.java]: Codacy Issue found: ['23' is a magic number.](https://www.codacy.com/app/calebepb/substituicao/file/2734585602/issues/source?bid=3308597&fileBranchId=3351375#l58)

**Comment by calebepb** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 7
- Added 3
           
```


See the complete overview on [Codacy](https://www.codacy.com/app/calebepb/substituicao/pullRequest?prid=254847&bid=3351375)

File: core/src/servico/NotificacaoService.java (hunk 1 above)
Comment (calebepb) [inline core/src/servico/NotificacaoService.java]: Codacy Issue found: [A catch statement that catches an exception only to rethrow it should be avoided.](https://www.codacy.com/app/calebepb/substituicao/file/2734585602/issues/source?bid=3308597&fileBranchId=3351375#l59)

File: core/src/servico/NotificacaoService.java (hunk 3 above)
Comment (calebepb) [inline core/src/servico/NotificacaoService.java]: Codacy Issue found: [Line is longer than 80 characters (found 94).](https://www.codacy.com/app/calebepb/substituicao/file/2734585602/issues/source?bid=3308597&fileBranchId=3351375#l58)

File: core/src/servico/NotificacaoService.java (hunk 2 above)
Comment (calebepb) [inline core/src/servico/NotificacaoService.java]: Codacy Issue found: [Line is longer than 80 characters (found 98).](https://www.codacy.com/app/calebepb/substituicao/file/2734585602/issues/source?bid=3308597&fileBranchId=3351375#l57)

File: core/src/servico/NotificacaoService.java (hunk 3 above)
Comment (calebepb) [inline core/src/servico/NotificacaoService.java]: Codacy Issue found: ['23' is a magic number.](https://www.codacy.com/app/calebepb/substituicao/file/2734585602/issues/source?bid=3308597&fileBranchId=3351375#l58)

**Comment by calebepb** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 4
- Added 4
           
```


See the complete overview on [Codacy](https://www.codacy.com/app/calebepb/substituicao/pullRequest?prid=254847&bid=3351375)

File: core/src/servico/NotificacaoService.java (hunk 1 above)
Comment (calebepb) [inline core/src/servico/NotificacaoService.java]: Codacy Issue found: [A catch statement that catches an exception only to rethrow it should be avoided.](https://www.codacy.com/app/calebepb/substituicao/file/2734585602/issues/source?bid=3308597&fileBranchId=3351375#l59)

File: core/src/servico/NotificacaoService.java (hunk 3 above)
Comment (calebepb) [inline core/src/servico/NotificacaoService.java]: Codacy Issue found: [Line is longer than 80 characters (found 94).](https://www.codacy.com/app/calebepb/substituicao/file/2734585602/issues/source?bid=3308597&fileBranchId=3351375#l58)

File: core/src/servico/NotificacaoService.java (hunk 2 above)
Comment (calebepb) [inline core/src/servico/NotificacaoService.java]: Codacy Issue found: [Line is longer than 80 characters (found 98).](https://www.codacy.com/app/calebepb/substituicao/file/2734585602/issues/source?bid=3308597&fileBranchId=3351375#l57)

File: core/src/servico/NotificacaoService.java (hunk 3 above)
Comment (calebepb) [inline core/src/servico/NotificacaoService.java]: Codacy Issue found: ['23' is a magic number.](https://www.codacy.com/app/calebepb/substituicao/file/2734585602/issues/source?bid=3308597&fileBranchId=3351375#l58)

**Comment by calebepb** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 4
- Added 4
           
```


See the complete overview on [Codacy](https://www.codacy.com/app/calebepb/substituicao/pullRequest?prid=254847&bid=3351375)

**Comment by calebepb** [comment]
[rebase, solve conflicts and push --force](https://github.com/Prof-Calebe/substituicao/blob/dev/README.md)
Fix problems from codacy - keep line length < 120
Do not remove this pull-request

**Comment by william-wc** [comment]
### Codacity: "**23** is a magic number"
https://en.wikipedia.org/wiki/Magic_number_(programming)

_"Unix or Linux scripts may start with a "shebang" (#!, **23** 21) followed by the path to an interpreter, if the interpreter is likely to be different from the one from which the script was invoked."_

Não é para este caso.

File: core/src/servico/NotificacaoService.java [hunk 4]
```diff
@@ -45,38 +45,23 @@ public String notificarAusencia(Long idProfessor, String dataInicio, String data
...
-            inícioReal = new DateTime(inicio.getYear(),inicio.getMonthOfYear(),inicio.getDayOfMonth(),inicio.getHourOfDay(),inicio.getMinuteOfHour());
-
             fim = new DateTime(sdf.parse(dataFim));
-            fimReal = new DateTime(fim.getYear(),fim.getMonthOfYear(),fim.getDayOfMonth(),fim.getHourOfDay(),fim.getMinuteOfHour());
+        } catch (ParseException pe) {
+            sdf = new SimpleDateFormat("dd/MM/yyyy");
+            inicio = new DateTime(sdf.parse(dataInicio)).withHourOfDay(0)
+                    .withMinuteOfHour(0);
+            fim = new DateTime(sdf.parse(dataFim)).withHourOfDay(23)
+                    .withMinuteOfHour(59);
```
Comment (calebepb) [inline core/src/servico/NotificacaoService.java]: Codacy Issue found: ['59' is a magic number.](https://www.codacy.com/app/calebepb/substituicao/file/2779878176/issues/source?bid=3364462&fileBranchId=3364495#l59)

File: core/src/servico/NotificacaoService.java [hunk 5]
```diff
@@ -45,38 +45,23 @@ public String notificarAusencia(Long idProfessor, String dataInicio, String data
...
             inicio = new DateTime(sdf.parse(dataInicio));
-            inícioReal = new DateTime(inicio.getYear(),inicio.getMonthOfYear(),inicio.getDayOfMonth(),inicio.getHourOfDay(),inicio.getMinuteOfHour());
-
             fim = new DateTime(sdf.parse(dataFim));
-            fimReal = new DateTime(fim.getYear(),fim.getMonthOfYear(),fim.getDayOfMonth(),fim.getHourOfDay(),fim.getMinuteOfHour());
+        } catch (ParseException pe) {
+            sdf = new SimpleDateFormat("dd/MM/yyyy");
+            inicio = new DateTime(sdf.parse(dataInicio)).withHourOfDay(0)
+                    .withMinuteOfHour(0);
+            fim = new DateTime(sdf.parse(dataFim)).withHourOfDay(23)
```
Comment (calebepb) [inline core/src/servico/NotificacaoService.java]: Codacy Issue found: ['23' is a magic number.](https://www.codacy.com/app/calebepb/substituicao/file/2779878176/issues/source?bid=3364462&fileBranchId=3364495#l58)

**Pull request merged**